.github/actions/deploy/action.yaml

```diff
@@ -0,0 +1,37 @@
+name: Deploy to Environment
+description: Build Docker Image, Push it to ECR
+
+permissions:
+  id-token: write
+  contents: read
+inputs:
+  docker-repo:
+    description: ECR Docker Repo to Push to
+    required: true
+  env-name:
+    description: One of 'prod' or 'staging'
+    required: true
+  role-to-assume:
+    description: AWS_ROLE_ARN
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        role-to-assume: ${{ inputs.role-to-assume }}
+        aws-region: us-east-1
+    - name: Build and Push Docker Image
+      uses: mbta/actions/build-push-ecr@v2
+      id: build-push-ecr
+      with:
+        docker-repo: ${{ inputs.docker-repo }}
+    - name: Deploy to ECS
+      uses: mbta/actions/deploy-ecs@v2
```

Comment:
i looked into this action a bit.

https://github.com/mbta/actions/blob/bb8457c70e8feccf57deb4f0d4a970beef4f2647/deploy-ecs/deploy.sh#L102 checks the status of the current service to decide if an existing process has been spun down and the new one has been spun up.

when running
`aws ecs describe-services --cluster dmap-import --services dmap-import-staging`
locally, i see a the portions of json (picked up [here](https://github.com/mbta/actions/blob/bb8457c70e8feccf57deb4f0d4a970beef4f2647/deploy-ecs/deploy.sh#L27-L30)) all being set to zero. this should cause this [while loop](https://github.com/mbta/actions/blob/main/deploy-ecs/deploy.sh#L100) to exit.

the setup and teardown portion of the action should work as expected with this configuration.

Reply:
Yes, but https://github.com/mbta/actions/blob/bb8457c70e8feccf57deb4f0d4a970beef4f2647/deploy-ecs/deploy.sh#L28 will wait for at least 1 task to be running.